### DISCUSSION TITLE
How to do abc ?? parent:abc with | extra1 | extra2

### DISCUSSION THREAD

**Discussion by niloc7** [statamic/cms#4748] [Q&A] [answered]
Hi guys!

`{{ code ?? parent:code | first:2 | upper }}`

When I do this, only the parent:code will be affected by the first:2 and upper..
When I put a bracket on ( code ?? parent:code ), an error appears.

Any idea how to go about this?

**Comment by jackmcdade** [comment, marked as answer]
Unfortunately duplication is the only way forward!

```
{{ code | first:2 | upper ?? parent:code | first:2 | upper }}
```

Try it in the browser: https://run.antlers.dev/lMXgCManW7kv8Gdj

**Reply by niloc7** [reply]
Ahh I see, thanks!!
